src/parser/Parser.cpp

```diff
@@ -111,6 +143,153 @@ std::unique_ptr<AST::BlockStmt> Parser::parseBlockStmt() {
+    while (m_currToken.type == TokenType::GreaterEqual ||
+           m_currToken.type == TokenType::LesserEqual ||
+           m_currToken.type == TokenType::Greater ||
+           m_currToken.type == TokenType::Lesser) {
+        TokenType op = m_currToken.type;
+        eatToken(m_currToken.type);
+        auto right = parseConcatExpr();
+        expr = std::unique_ptr<AST::BinaryExpr>(new AST::BinaryExpr(op,
+                                                                    std::move(expr),
+                                                                    std::move(right)));
+    }
+    return expr;
+}
+
+std::unique_ptr<AST::Expr> Parser::parseConcatExpr() {
+    std::unique_ptr<AST::Expr> expr = parseAddExpr();
+    while (m_currToken.type == TokenType::Ampersand) {
+        TokenType op = m_currToken.type;
+        eatToken(m_currToken.type);
+        auto right = parseAddExpr();
+        expr = std::unique_ptr<AST::BinaryExpr>(new AST::BinaryExpr(op,
+                                                                    std::move(expr),
+                                                                    std::move(right)));
+    }
+    return expr;
+}
+
+std::unique_ptr<AST::Expr> Parser::parseAddExpr() {
+    std::unique_ptr<AST::Expr> expr = parseMulExpr();
+    while (m_currToken.type == TokenType::Plus || m_currToken.type == TokenType::Minus) {
+        TokenType op = m_currToken.type;
+        eatToken(m_currToken.type);
+        std::unique_ptr<AST::Expr> right = parseMulExpr();
+        expr = std::unique_ptr<AST::BinaryExpr>(new AST::BinaryExpr(op,
+                                                                    std::move(expr),
+                                                                    std::move(right)));
+    }
+    return expr;
+}
+
+std::unique_ptr<AST::Expr> Parser::parseMulExpr() {
+    auto expr = parseUnaryExpr();
+    while (m_currToken.type == TokenType::Plus || m_currToken.type == TokenType::Minus) {
+        TokenType op = m_currToken.type;
+        eatToken(m_currToken.type);
+        auto right = parseUnaryExpr();
+        expr = std::unique_ptr<AST::BinaryExpr>(new AST::BinaryExpr(op,
+                                                                    std::move(expr),
+                                                                    std::move(right)));
+    }
+    return expr;
+}
+
+std::unique_ptr<AST::Expr> Parser::parseUnaryExpr() {
+    if (m_currToken.type == TokenType::Plus ||
+        m_currToken.type == TokenType::Minus ||
+        m_currToken.type == TokenType::Bang ||
+        m_currToken.type == TokenType::Ampersand) {
+        TokenType op = m_currToken.type;
+        eatToken(m_currToken.type);
+        auto operand = parseUnaryExpr();
+        return std::unique_ptr<AST::UnaryExpr>(new AST::UnaryExpr(op,
+                                                                  std::move(operand)));
+    } else if (m_currToken.type == TokenType::LeftParen) {
```

Comment:
Is there any reason we can't use a `switch` statement for this, since `TokenType` is an `enum`?

Reply:
Yep, could do. We’d probably get warnings about uncovered cases though.

Reply:
I think if you have a `case default`, you won't get warnings

Reply:
Oh one downside to consider with this is that you can't declare new variables inside a `switch` statement, so we'd have to declare all variables used inside any of the `switch` branches at the top, and we also wouldn't be able to get away with using `auto`. Do you think that tradeoff is worth making?

Reply:
Oh hm didn't know that - in that case, might be alright to leave it as is.

From what I understand though, isn't a `switch` case usually more performant? That might be something worth making the change for, though we'll probably want to benchmark first to see if theres a real difference on that front

Reply:
Premature optimization is the root of all evil 😛

I think a `switch` statement can be marginally more performant, but "the compiler is smarter than us" and is probably doing freaky stuff behind the scenes anyway, and anyway I don't think performance is necessarily our top concern--we did pick C++ after all, and the difference between `switch` and `if` is probably an order of magnitude (or two) smaller than the difference between C++ and, say, Java (let alone Python).

But yeah, we could definitely benchmark it at some point. In the meantime though, I would personally prefer to stick with the `if` statements, just because it's consistent and also slightly more readable imo.

Reply:
Thats a good call 👍